### PR TITLE
QPPA-2180: Add Developer Preview content

### PR DESCRIPTION
**Pull request by marimiyachi**
https://jira.cms.gov/browse/QPPA-2180

Developer Preview content changes can be viewed [here](https://drive.google.com/file/d/1PuNTENawOgSkFuhQWoRcr_uujbJAS42t/view?usp=sharing). Introduction is [here](https://drive.google.com/file/d/1chvXyNnik96W12ApgkGzJmzM0e-NZagy/view?usp=sharing).

Note: "Get an API Key" content is a placeholder.

### CODE REVIEW

**Comment by ivanang** [comment]
'Get an API key' is subject to change pending mtg with Ad Hoc on 8/23. So we shouldn't push this PR to prod until that's settled. Or we can break that out as a separate PR.

File: src/components/developer-preview.js [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+              <td></td>
+              <td>Same as /measurement-sets</td>
+            </tr>
+          </tbody>
+        </table>
+
+        <h1 className='ds-h1' id='endpoints'>Endpoints</h1>
+        <p className='ds-text'>Visit the <a href='https://qpp-submissions-sandbox.navapbc.com/#!'>Interactive Documentation</a> to learn more about the endpoints available in the Submissions API.</p>
+        <p className='ds-text'>If you are a Registry or QCDR using the Developer Preview, you have an API key that is associated with an 'organization'. This affects what endpoints you are authorized to use, and what behavior each endpoint has.</p>
+        <p className='ds-text'>At a high level, your API key allows you to create and edit data using the /measurement-sets and /measurements endpoints. You cannot create data using the /submissions endpoint. For more information about what you're authorized to do with the Submissions API, click <a href='/qpp-submissions-docs/developer-preview#authorization'>here</a>.</p>
```
Comment (ivanang) [inline src/components/developer-preview.js]: should we make /measurement-sets , '/measurements and '/submissions' a different text to make it clear that those are API endpoints?

**Comment by ivanang** [comment]
Also, does PR include the content update to the 'Introduction' page?

**Comment by marimiyachi** [comment]
@ivanang it does - I've added a PDF with the intro. I also addressed your comments!

**Comment by marimiyachi** [comment]
Get API key content will be added here: https://jira.cms.gov/browse/QPPA-1600

**Comment by marimiyachi** [comment]
Thanks @ivanang - do you feel comfortable approving?

File: src/components/developer-preview.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        </ul>
+
+        <h1 className='ds-h1' id='api-key'>Get an API Key</h1>
+        <p className='ds-text'>This content is not yet available. Please check back later.</p>
+
+        <h1 className='ds-h1' id='authentication'>Authentication</h1>
+        <p className='ds-text'>You must authenticate your account when using the <a href='https://cmsgov.github.io/qpp-submissions-docs/developer-preview'>Developer Preview</a> (base url: https://preview.qpp.cms.gov/api/submissions/). Authenticate via bearer auth by adding your API token to the header of every request using the key value: <strong>Authorization: Bearer [YOUR API TOKEN]</strong>.</p>
+        <p className='ds-text'>API keys carry many privileges, and must not be shared in publicly accessible areas such as GitHub and in client-side code.  Even within organizations, access must be limited to staff embedding it in software.</p>
+        <p className='ds-text'>Your API key carries many privileges, so be sure to keep it secret! Do not share your secret API key in publicly accessible areas such GitHub, client-side code, and so forth.</p>
```
Comment (kencheeto) [inline src/components/developer-preview.js]: these two paragraphs are repetitive, we could just keep one

File: src/components/developer-preview.js [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+
+        <h2 className='ds-h2'>Notes</h2>
+        <ul>
+          <li>Any data submitted in the Developer Preview will not count toward a MIPS-eligible clinician or group's final score in the Quality Payment Program.</li>
+          <li>Since the Developer Preview is an integration environment for the Submissions API, it may be changed in ways that break backward compatibility and is not subject to any service-level guarantees or deprecation policy.</li>
+        </ul>
+
+        <h2 className='ds-h2'>Where to go for support</h2>
+        <ul>
+          <li>Send test API requests in the <a href='https://qpp-submissions-sandbox.navapbc.com/'>Interactive Documentation</a> using your Developer Preview API key.</li>
```
Comment (ivanang) [inline src/components/developer-preview.js]: can we make this link open in a new tab?

File: src/components/introduction.js [hunk 4]
```diff
@@ -7,36 +7,20 @@ class Introduction extends React.Component {
...
-          <li>POST https://qpp.cms.gov/api/submissions/measurement-sets</li>
-          <li>PUT, PATCH, POST, DELETE https://qpp.cms.gov/api/submissions/measurement-sets/<span>&#123;id&#125;</span></li>
-        </ul>
-        Measurements
-        <ul>
-          <li>POST https://qpp.cms.gov/api/submissions/measurements</li>
-          <li>PUT, PATCH, POST, DELETE https://qpp.cms.gov/api/submissions/measurements/<span>&#123;id&#125;</span></li>
-        </ul>
+        <h2 className='ds-h2'>Explore and integrate with the Submissions API</h2>
+        <p className='ds-text'>To make the API as explorable as possible, we have <a href='https://qpp-submissions-sandbox.navapbc.com/'>Interactive Documentation</a>, which not only documents the endpoints available but also allows you to make requests to the API and see what response you would get. The Interactive Documentation is connected to the Developer Preview, which is an integration environment for registries and QCDRs.</p>
```
Comment (ivanang) [inline src/components/introduction.js]: make the link open in a new tab

File: src/components/introduction.js [hunk 5]
```diff
@@ -7,36 +7,20 @@ class Introduction extends React.Component {
...
-        <p className='ds-text'>GET https://qpp.cms.gov/api/submissions/submissions/<span>&#123;id&#125;</span>/score will also remain accessible. Please note that this endpoint returns the score for a submission object, and does not represent the final score for a TIN/NPI.</p>
         <h2 className='ds-h2'>Stay Up Date</h2>
-        <p className='ds-text'>The QPP APIs Google Group is where developers and the CMS QPP Product Managers discuss your questions and feedback. New announcements and discussions are being added everyday.</p>
+        <p className='ds-text'>The <a href='https://groups.google.com/forum/#!forum/qpp-apis'>QPP APIs Google Group</a> is where developers and the CMS QPP Product Managers discuss your questions and feedback. New announcements and discussions are being added everyday.</p>
         <p className='ds-text'>For questions about QPP measures, scores, policy or how the program works, please contact the Quality Payment Program Service Center at QPP@cms.hhs.gov or 1-866-288-8292.</p>
```
Comment (ivanang) [inline src/components/introduction.js]: make QPP@cms.hhs.gov an email link

File: src/components/developer-preview.js [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+
+import SubmissionsObjects from './submissions-objects';
+
+class DeveloperPreview extends React.Component {
+  render() {
+    return (
+      <div>
+        <h1 className='ds-h1'>Developer Preview</h1>
+        <ul>
+          <li><a href='#overview'>Overview</a></li>
```
Comment (ivanang) [inline src/components/developer-preview.js]: when I click on these links, they jump to the start of the paragraph _below_ the heading/title. is it possible for the link to jump to the heading/title?

File: src/components/developer-preview.js [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+          <li><a href='#api-key'>Get an API Key</a></li>
+          <li><a href='#authentication'>Authentication</a></li>
+          <li><a href='#authorization'>Authorization</a></li>
+          <li><a href='#endpoints'>Endpoints</a></li>
+          <li><a href='#special-scoring'>How to Test Special Scoring Scenarios</a></li>
+        </ul>
+
+        <h1 className='ds-h1' id='overview'>Overview</h1>
+        <p className='ds-text'>The Developer Preview allows developers to build direct integrations with other software to test programmatically submitting Quality Payment Program (QPP) performance data to CMS. The API’s functionality in the Developer Preview closely matches that of the QPP system that is in production.</p>
+        <p className='ds-text'>The base URL for the Developer Preview is: https://preview.qpp.cms.gov/api/submissions/</p>
```
Comment (ivanang) [inline src/components/developer-preview.js]: bold https://preview.qpp.cms.gov/api/submissions

File: src/components/developer-preview.js [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+        </ol>
+
+        <h2 className='ds-h2'>Resources</h2>
+        <ul>
+          <li>Review the <a href='https://www.cms.gov/Medicare/Quality-Payment-Program/Resource-Library/Enterprise-Identity-Data-Management-EIDM-User-Guide.pdf'>EIDM User Guide</a></li>
+          <li>Contact the Quality Payment Program at QPP@cms.hhs.gov or 1-866-288-8292/TTY: 1-877-715-6222.</li>
+        </ul>
+
+        <h1 className='ds-h1' id='authentication'>Authentication</h1>
+        <p className='ds-text'>You must authenticate your account when using the <a href='https://cmsgov.github.io/qpp-submissions-docs/developer-preview'>Developer Preview</a> (base url: https://preview.qpp.cms.gov/api/submissions/). Authenticate via bearer auth by adding your API token to the header of every request using the key value: Authorization: Bearer [YOUR API TOKEN].</p>
```
Comment (ivanang) [inline src/components/developer-preview.js]: remove the link for 'Developer Preview' in line 97. Bold 'https://preview.qpp.cms.gov/api/submissions/'

File: src/components/developer-preview.js (hunk 2 above)
Comment (ivanang) [reply src/components/developer-preview.js]: yup, good call out @kencheeto . @marimiyachi  - delete the 2nd paragraph.

File: src/components/developer-preview.js [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+        <p className='ds-text'>Your API key carries many privileges, so be sure to keep it secret! Do not share your secret API key in publicly accessible areas such GitHub, client-side code, and so forth.</p>
+        <p className='ds-text'>All API requests must be made over HTTPS. Calls made over plain HTTP will fail. API requests in the Developer Preview without authentication will also fail.</p>
+        <p className='ds-text'>For instructions on how to get an API token, click <a href='/qpp-submissions-docs/developer-preview#api-key'>here</a>.</p>
+
+        <h1 className='ds-h1' id='Authorization'>Authorization</h1>
+        <p className='ds-text'>If you are a Registry or QCDR using the Developer Preview, you have an API key that is associated with an 'organization'. This affects what endpoints you are authorized to use, and what behavior each endpoint has.</p>
+        <p className='ds-text'>In general, if you have an organization-type API key, you can:</p>
+        <ul>
+          <li>Submit data for any TIN without any prior certification</li>
```
Comment (ivanang) [inline src/components/developer-preview.js]: delete 'without any prior certification'. might confuse users who say 'hey wait arent i certified b/c i am a registry?'

File: src/components/developer-preview.js [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+              <th>Endpoint</th>
+              <th>HTTP Method</th>
+              <th>Description</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td>/submissions</td>
+              <td>GET</td>
+              <td>Returns all submissions that you have submitted measurement sets for, and within those submissions, returns only the measurement sets that you have  submitted. If qpp-taxpayer-identification-number is specified in the header as a query parameter, and you haven't submitted any measurement sets for the specified TIN, this endpoint returns a submission object with an empty measurement sets array.</td>
```
Comment (ivanang) [inline src/components/developer-preview.js]: should we add <code> tags around 'qpp-taxpayer-identification-number'?

File: src/components/developer-preview.js [hunk 11]
```diff
@@ -0,0 +1,288 @@
...
+
+        <h1 className='ds-h1' id='special-scoring'>How to Test Special Scoring Scenarios</h1>
+        <p className='ds-text'>A submission to the Submissions API requires a Taxpayer Identification Number (TIN) and may include a National Provider Identifier (NPI). These identifiers are used to accurately attribute the submission to a provider. A provider's profile includes information which is used to trigger special scoring scenarios.</p>
+        <p className='ds-text'>Currently, no special scoring scenarios are applied in the production environment. Every submission is attributed with the default provider profile, which incurs no special scoring.</p>
+        <p className='ds-text'>In order to support the testing of special scoring scenarios, the Submissions API features a provider profile stub in the Developer Preview environment for the /score-preview endpoint. The provider profile stub is not applied to the /score endpoint.</p>
```
Comment (ivanang) [inline src/components/developer-preview.js]: add code tags to /score-preview and /score

File: src/components/developer-preview.js [hunk 12]
```diff
@@ -0,0 +1,288 @@
...
+            <li>Measurements: A measurement object represents one single data point related to a specific measure in a given category, and is tied to a measurement set object.</li>
+          </ul>
+        </p>
+
+        <h2 className='ds-h2'>Authentication</h2>
+        <p className='ds-text'>Both the Developer Preview and the Production environments require a JSON Web Token ("API key") for authentication purposes.</p>
+
+        <a className='ds-c-button ds-c-button--primary' href='/qpp-submissions-docs/developer-preview#api-key'>Get an API Key</a>
+
+        <h2 className='ds-h2'>Notes</h2>
```
Comment (ivanang) [inline src/components/developer-preview.js]: Move this section and 'where to go for support' to the bottom of this page

**Pull request merged**